src/compiler/checker.ts

```diff
@@ -2623,8 +2623,9 @@ namespace ts {
                         }
                         if (!context.encounteredError && !(context.flags & NodeBuilderFlags.AllowEmptyTuple)) {

```

Comment:
Invert the predicate here to make it more readable and put `return undefined` at the end.